src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/repository/TaskResourceRepositoryTest.java

```diff
@@ -233,17 +391,167 @@ void given_task_is_created_when_count_by_role_signature_then_correct_count_retur
+
+    @Test
+    void given_tasks_created_when_search_request_without_state_filter_then_return_only_assigned_and_unassigned_tasks() {
+        String taskId2 = UUID.randomUUID().toString();
+        TaskResource createdTask = createTask(taskId2, "case-manager", "IA",
+            "reviewAppeal", "anotherAssignee", "1623278362430413", CFTTaskState.UNASSIGNED);
+
+        transactionHelper.doInNewTransaction(() -> {
+            task.setIndexed(true);
+            createdTask.setIndexed(true);
+            taskResourceRepository.save(task);
+            taskResourceRepository.save(createdTask);
+        });
+
+        Set<String> filterSignature = Set.of("*:IA:*:*:1:765324");
+        Set<String> roleSignature = Set.of("IA:*:*:tribunal-caseofficer:*:r:U:*", "IA:*:*:case-manager:*:r:U:*");
+        SearchRequest request = SearchRequest.builder()
+            .caseIds(List.of("1623278362430412", "1623278362430413"))
+            .taskTypes(List.of("startAppeal", "reviewAppeal"))
+            .users(List.of("someAssignee", "anotherAssignee"))
+            .sortingParameters(List.of(new SortingParameter(SortField.CASE_ID, SortOrder.ASCENDANT)))
+            .build();
+
+        List<String> taskIds = taskResourceRepository.searchTasksIds(0,25, filterSignature, roleSignature,
+            List.of(), request);
+        assertEquals(2, taskIds.size());
+        assertEquals(taskId, taskIds.get(0));
+        assertEquals(taskId2, taskIds.get(1));
+    }
+
+    @Test
+    void given_tasks_created_when_search_request_with_excluded_case_id_then_tasks_from_excluded_id_is_not_returned() {
+        String taskId2 = UUID.randomUUID().toString();
+        TaskResource createdTask = createTask(taskId2, "case-manager", "IA",
+            "reviewAppeal", "anotherAssignee", "1623278362430413", CFTTaskState.COMPLETED);
+
+        transactionHelper.doInNewTransaction(() -> {
+            task.setIndexed(true);
+            createdTask.setIndexed(true);
+            taskResourceRepository.save(task);
+            taskResourceRepository.save(createdTask);
+        });
 
-        Long taskCount = taskResourceRepository.searchTasksCount(filterSignature, roleSignature);
+        Set<String> filterSignature = Set.of("*:IA:*:*:1:765324");
+        Set<String> roleSignature = Set.of("IA:*:*:tribunal-caseofficer:*:r:U:*", "IA:*:*:case-manager:*:r:U:*");
+        SearchRequest request = SearchRequest.builder()
+            .cftTaskStates(List.of(CFTTaskState.ASSIGNED, CFTTaskState.COMPLETED))
+            .caseIds(List.of("1623278362430412", "1623278362430413"))
+            .taskTypes(List.of("startAppeal", "reviewAppeal"))
+            .users(List.of("someAssignee", "anotherAssignee"))
+            .build();
+
+        List<String> taskIds = taskResourceRepository.searchTasksIds(0,25, filterSignature, roleSignature,
+            List.of("1623278362430413", "88888888888888"), request);
+        assertEquals(1, taskIds.size());
+        assertEquals(taskId, taskIds.get(0));
```

Comment:
assert tasks do not have excluded id?

Reply:
Done in RWA-2212 in CFTTaskDatabaseServiceSearchTest